runtime/tr.source/trj9/runtime/ClassUnloadAssumption.cpp

```diff
@@ -183,32 +183,30 @@ OMR::RuntimeAssumption::dumpInfo(char *subclassName)
 
 bool OMR::RuntimeAssumption::isAssumingRange(uintptrj_t rangeStartPC, uintptrj_t rangeEndPC, uintptrj_t rangeColdStartPC, uintptrj_t rangeColdEndPC, uintptrj_t rangeStartMD, uintptrj_t rangeEndMD)
    {
-   uintptrj_t pc = (uintptrj_t) getAssumingPC();
-   return (pc >= rangeStartPC && pc <= rangeEndPC ||
-           (rangeColdStartPC && pc >= rangeColdStartPC && pc <= rangeColdEndPC) ||
-           (rangeStartMD && pc >= rangeStartMD && pc <= rangeEndMD));
+   return (assumptionInRange(rangeStartPC, rangeEndPC) ||
+           (rangeColdStartPC && assumptionInRange(rangeColdStartPC, rangeColdEndPC)) ||
+           (rangeStartMD && assumptionInRange(rangeStartMD, rangeEndMD)));
    }
 
 bool OMR::RuntimeAssumption::isAssumingMethod(void *md, bool reclaimPrePrologueAssumptions)
    {
    J9JITExceptionTable *metaData = (J9JITExceptionTable*) md;
-   uintptrj_t pc = (uintptrj_t) getAssumingPC();
 
    // MetaData->startPC refers to the interpreter entry point, and does not include pre-prologue.
    // If we need to reclaim pre-prologue assumptions (i.e. on class unloading), we need to scan
    // from the beginning of the code cache allocation of the method (metaData->codeCacheAlloc).
-   uintptrj_t startPC = (reclaimPrePrologueAssumptions)?metaData->codeCacheAlloc:metaData->startPC;
-   if (pc >= startPC && pc <= metaData->endWarmPC ||
-           (metaData->startColdPC && pc >= metaData->startColdPC && pc <= metaData->endPC))
+   uintptrj_t metaStartPC = (reclaimPrePrologueAssumptions)?metaData->codeCacheAlloc:metaData->startPC;
+   if (assumptionInRange(metaStartPC, metaData->endWarmPC) ||
+           (metaData->startColdPC && assumptionInRange(metaData->startColdPC, metaData->endPC)))
       return true;
 
-   if ((pc >= (uintptrj_t) metaData) && (pc <= (((uintptrj_t) metaData) + ((uintptrj_t) metaData->size))))
+   if (assumptionInRange((uintptrj_t)metaData, (uintptrj_t)(metaData + metaData->size)))
```

Comment:
`metaData + metaData->size` is now pointer arithmetic, adding `metaData->size * sizeof (*metaData)` to the address. Is that intentional?

Reply:
Thanks, this was a mistake in converting the range check. It has been corrected.